src/main/daml/Daml/Finance/Interface/Account/Account.daml

```diff
@@ -178,21 +164,31 @@ undisclose observersToRemove actor disclosers account =
       actor
       Disclosure.RemoveObservers with disclosers; observersToRemove
 
+-- | Exercise interface by key.
+-- This method can be used to exercise a choice on an `Account` given its `AccountKey`.
+-- Requires as input the `AccountKey`, the actor fetching the account and the choice arguments.
+-- For example:
+-- ```
+-- exerciseInterfaceByKey @Account.I accountKey actor Account.Debit with holdingCid
+-- ```
+exerciseInterfaceByKey : forall i c r.
```

Comment:
There might be a way to define a type-class linking 
- the key type (`AccountKey`)
- the interface type (`Account`)
- the reference type (`Reference`)

and implement an `exerciseInterfaceByKey` method which gets defined only once. I think the trick is how to abstract the `GetCid with viewer` piece ...

Reply:
I found a way to do it

Reply:
I attempted to consolidate the type annotations. However, doing so without adding either

```
{-# LANGUAGE MultiParamTypeClasses #-}
```

or

```
{-# LANGUAGE TypeFamilies #-}
```
proves to be challenging. Both extensions lead to warnings, such as:

```
Modules compiled with the TypeFamilies language extension may encounter issues with data-dependencies.
```

As a workaround, I tried to introduce a dummy input parameter to `exerciseInterfaceByKey` with the type

```
data Bundle i v k r c1 c2 c3 c4 = Bundle {}
```

The sole purpose of that argument being to pass the types, and while effective, it's not the most elegant solution. 

Hence, I opted for bespoke `exerciseByInterfaceKey` functions with appropriate type annotations directly within their respective modules: `Account.I`, `Instrument.I`, and `HoldingFactory.I`.Hopefully, this version reads smoother.